### PR TITLE
Add openai api base support

### PR DESCRIPTION
**Pull request by cnych**
### Background
In some scenarios, it is not possible to directly access the official OpenAI API address. It is necessary to access it through a proxy address, which can be specified by specifying env `OPENAI_API_BASE`.

### Changes
Added an `OPENAI_API_BASE` environment variable to configure the base URL of the OpenAI API.

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Review by nponeccop — DISMISSED** [review]
OPENAI_API_BASE is never used

**Comment by cnych** [comment]
> OPENAI_API_BASE is never used

You can override the default `https://api.openai.com/v1` by configuring `OPENAI_API_BASE` in `.env`.

This is very much needed for scenarios where direct access to the OpenAI API is not possible.

**Review by zsjinwei — DISMISSED** [review]
test pass

**Comment by cryptidv** [comment]
> > It's not actually used when making API requests.
> > Additionally, wouldn't this violate OpenAI's TOS? If you can't access OAI's APIs either due to geoblocking or some other reason, can't you just use a VPN (or even Azure)?
> 
> In fact, the OpenAI SDK itself provides a direct configuration entry for `openai.api_base`. For example, you can use Cloudflare Worker service to proxy the OpenAI API and perform related statistics at the Cloudflare level.

Ah I see ok! LGTM then 👍

**Comment by nponeccop** [comment]
@cnych There are conflicts now

**Comment by cnych** [comment]
> @cnych There are conflicts now

The conflict has been resolved.

**Comment by nponeccop** [comment]
@cnych We decided to do another major cleanup, so that the code is very tidy. It will be ready later today so I'll notify when it's time to fix conflicts again. And, if you happen to speak Chinese, can you join our discussion in #1508 ?

**Comment by cnych** [comment]
> @cnych We decided to do another major cleanup, so that the code is very tidy. It will be ready later today so I'll notify when it's time to fix conflicts again. And, if you happen to speak Chinese, can you join our discussion in #1508 ?

Ok！I'm Chinese! I'll take a look at your discussion.

**Comment by nponeccop** [comment]
@cnych Thanks. There are conflicts now.

**Comment by gaye746560359** [comment]
> 

如何设置？我试了在。env文件添加OPENAI_API_BASE=https://sssssss/v1并没有效果

**Comment by cnych** [comment]
> > 
> 
> 如何设置？我试了在。env文件添加OPENAI_API_BASE=[https://sssssss/v1并没有效果](https://sssssss/v1%E5%B9%B6%E6%B2%A1%E6%9C%89%E6%95%88%E6%9E%9C)

You need add `openai.api_base = os.getenv("OPENAI_API_BASE", "https://api.openai.com/v1")` to `config.py`.

**Comment by gaye746560359** [comment]
> > > 
> > 
> > 
> > 如何设置？我试了在。env文件添加OPENAI_API_BASE=[https://sssssss/v1并没有效果](https://sssssss/v1%E5%B9%B6%E6%B2%A1%E6%9C%89%E6%95%88%E6%9E%9C)
> 
> You need add `openai.api_base = os.getenv("OPENAI_API_BASE", "https://api.openai.com/v1")` to `config.py`.

有效，非常感谢

**Comment by nponeccop** [comment]
@cnych There are conflicts now

**Comment by cnych** [comment]
> @cnych There are conflicts now

Oh... could you merge this PR first? The update is too fast now, I have resolved four or five conflicts～

File: requirements-docker.txt
Comment (nponeccop) [inline requirements-docker.txt:1]: This looks unrelated. Revert.

**Review by nponeccop — CHANGES_REQUESTED** [review]

**Comment by sweetlilmre** [comment]
Hi. Suggest closing this PR in favour of #2594 which adds broader functionality

**Pull request closed without merging**